lib/Doctrine/DBAL/Connection.php

```diff
@@ -1207,10 +1207,19 @@ public function executeCacheQuery($sql, $params, $types, QueryCacheProfile $qcp)
             } elseif (array_key_exists($realKey, $data)) {
                 $stmt = new ArrayStatement([]);
             }
+        } else {
+            $data = [];
```

Comment:
Why is this needed?

Reply:
It is needed to distinguish "data from query cache is not loaded" and "data from query cache is loaded and empty". 
This data was be set to empty array in ResultCacheStatement before - https://github.com/doctrine/dbal/blob/2.11.x/lib/Doctrine/DBAL/Cache/ResultCacheStatement.php#L341